### PR TITLE
Update and rename lecture21.md to Day21Notes.md

### PR DESCRIPTION
**Pull request by jmbaumer**
- Update and fill in the Day21Notes.md markdown with the lecture notes for day 21
- The notes adhere to the style guide recommendation

### CODE REVIEW

File: Day21Notes.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+Steps to reproduce $\rightarrow$ minimized (also apply to input data)
+
+- Mozilla Engineers had lots of open bug reports that were not minimized
+- Mozilla ran a Bug-a-thon crowd sourced simplification
+
+Ex/ FF $\rightarrow$ print page causes a crash
+- What part of the HTML is causing the crash?
+
+Another problem where minimization ccan help:
```
Comment (jabelm20) [inline Day21Notes.md:40]: Fix Spelling of can.

File: Day21Notes.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+- Yesteray your code worked, today it does not
+- Why? - a number of commits
+  - which commit?
+
+>#### Definition
+>**Causality** is influence by which one evenet, process, state or object (a cause) contributes to the production of anothe event, process, state or object,   (an effect) where the cause is partly responsible for the effeect and the effect is partly dependent on the cause
```
Comment (jabelm20) [inline Day21Notes.md:46]: Fix spelling of event, another, effect.

File: Day21Notes.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+>#### Definition
+>**Counterfactual** theories of causation:
+>If A had not ccurred, then C would not have occurred
```
Comment (jabelm20) [inline Day21Notes.md:49]: Spelling: occurred

**Review by jabelm20 — CHANGES_REQUESTED** [review]
I added comments on the file about some spelling errors. But overall looks good with formatting. I also really liked the fun thing you added at the end!

**Comment by kevinaangstadt** [comment]
@jmbaumer please make the requested edits :-)

**Pull request merged**